File: wk3/kikiboba.java
// problem link: https://open.kattis.com/problems/kikiboba

import java.io.*;
import java.math.*;
import java.util.*;

public class kikiboba
{
	public static void main(String[] args) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		char[] s = r.readLine().toCharArray();
		int b = 0, k = 0;
		for(char c : s) {
		    if(c == 'b')
		        b++;
		    else if(c == 'k')
		        k++;
		}
		if(b == k) {
		    if(b == 0)
		        System.out.println("none");
		    else
		        System.out.println("boki");
		}
		else {
		    if(b > k)
		        System.out.println("boba");
		    else
		        System.out.println("kiki");
		    
		}
	}
}

File: wk3/cpp/kikiboba.cpp
// problem link: https://open.kattis.com/problems/kikiboba

#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    string s;
    int b = 0, k = 0;
    cin >> s;
    for(char& c : s) {
        if(c == 'b')
            b++;
        else if(c == 'k')
            k++;
    }
    if(b == k) {
        if(b == 0)
            cout << "none";
        else
            cout << "boki";
    }
    else {
        if(b > k)
            cout << "boba";
        else
            cout << "kiki";
    }
    
    return 0;
}